lib/Transforms/Scalar/UnifyAssumes.cc

```diff
@@ -143,7 +145,7 @@ bool UnifyAssumesPass::runOnFunction(Function &F) {
 
   // -- delete all assumes
   for (auto ci : assumes)
-    ci->eraseFromParent();
+    markAssumeAsUnified(*ci);
```

Comment:
there should be a flag to erase `ci` to preserve current behavior, just in case it is needed